django/forms/fields.py

```diff
@@ -315,12 +319,6 @@ def validate(self, value):
         if not math.isfinite(value):
             raise ValidationError(self.error_messages['invalid'], code='invalid')
 
-    def widget_attrs(self, widget):
-        attrs = super().widget_attrs(widget)
-        if isinstance(widget, NumberInput) and 'step' not in widget.attrs:
-            attrs.setdefault('step', 'any')
-        return attrs
-
```

Comment:
This introduces a regression from 7ec2a21be15af5b2c7513482c3bcfdd1e12782ed. 

We need a selenium test to cover it, but it I create an `<input type="number" />` without the `step` specified at all, I can't enter a non-integer value, such as `0.5` without triggering the browser validation (_Enter a valid value_). 

This ties in with [MDN docs for `step`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#step): 

> The default stepping value for number inputs is 1, allowing only integers to be entered…